src/daemon/pex/pex.go

```diff
@@ -215,6 +229,33 @@ func New(cfg Config, defaultConns []string) (*Pex, error) {
 	return pex, nil
 }
 
+func (px *Pex) download() error {
+	resp, err := http.Get(PeerDatabaseUrl)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	for _, address := range strings.Split(string(body), "\n") {
+		if validateAddress(address, px.Config.AllowLocalhost) {
+			err := px.AddPeer(address)
```

Comment:
Parse the peerlist into a list of addresses, then shuffle the list before adding